### PR TITLE
Add editevent command and its test cases

### PR DESCRIPTION
**Pull request by inas0ng**
This commit add editevent command and its test cases.

Note that editevent:
- Allows for user to edit an event's name, start date time, and/or start end time.
- Edits the relevant event tag tied to all persons in address book.
- Will detect end datetime < start datetime as an error if start/end datetime is edited to be of invalid range.

The UG for the editevent feature has also been added in this commit.

### CODE REVIEW

File: src/test/java/seedu/address/testutil/EditEventDescriptorBuilder.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
+package seedu.address.testutil;
+
+import seedu.address.logic.commands.EditEventCommand.EditEventDescriptor;
+import seedu.address.model.event.DateTime;
+import seedu.address.model.event.Event;
+import seedu.address.model.event.EventName;
+
+/**
+ * A utility class to help with building EditPersonDescriptor objects.
```
Comment (weilin1202) [inline src/test/java/seedu/address/testutil/EditEventDescriptorBuilder.java:9]: The java docs should be EditEventDescriptor

File: src/test/java/seedu/address/logic/commands/EditEventCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+        EditEventCommand editEventCommand = new EditEventCommand(INDEX_SECOND_EVENT, descriptor);
+
+        assertCommandFailure(editEventCommand, model, EditEventCommand.MESSAGE_DUPLICATE_EVENT);
+    }
+
+    @Test
+    public void execute_duplicateEventFilteredList_failure() {
+        showEventAtIndex(model, INDEX_FIRST_EVENT);
+
+        // edit person in filtered list into a duplicate in address book
```
Comment (weilin1202) [inline src/test/java/seedu/address/logic/commands/EditEventCommandTest.java:119]: edit 'event' in filtered list

File: src/main/java/seedu/address/model/AddressBook.java [hunk 3]
```diff
@@ -158,6 +159,44 @@ public void deleteEventFromPersonList(Event eventToDelete) {
         }
     }
 
+    /**
```
Comment (weilin1202) [inline src/main/java/seedu/address/model/AddressBook.java:162]: Maybe can add in "Updates the event set of person in address book" to make the function of method clearer

File: src/main/java/seedu/address/model/AddressBook.java [hunk 4]
```diff
@@ -158,6 +159,44 @@ public void deleteEventFromPersonList(Event eventToDelete) {
...
+                }
+                Person editedPerson = new Person(personToCheck.getName(), personToCheck.getPhone(),
+                        personToCheck.getEmail(), personToCheck.getAddress(), editedEventSet);
+                persons.setPerson(personToCheck, editedPerson);
+            }
+        }
+    }
+
+    /**
+     * Replaces the given person {@code target} in the list with {@code editedPerson}.
```
Comment (weilin1202) [inline src/main/java/seedu/address/model/AddressBook.java:190]: Remember to update the javadocs haha :)

**Review by weilin1202 — COMMENTED** [review]
Overall its good! I'm ok with the implementation and I just have some comments about the javadocs only :)

**Review by weilin1202 — APPROVED** [review]
It looks good to me! :)

File: src/test/java/seedu/address/testutil/EditEventDescriptorBuilder.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+     */
+    public EditEventDescriptorBuilder(Event event) {
+        descriptor = new EditEventDescriptor();
+        descriptor.setEventName(event.getName());
+        descriptor.setStartDateTime(event.getStartDateTime());
+        descriptor.setEndDateTime(event.getEndDateTime());
+    }
+
+    /**
+     * Sets the {@code eventName} of the {@code EditPersonDescriptor} that we are building.
```
Comment (venuslimm) [inline src/test/java/seedu/address/testutil/EditEventDescriptorBuilder.java:34]: I think there is a typo in the JavaDocs?

**Review by venuslimm — COMMENTED** [review]
The code looks good to me! I have left some comments about the documentation and JavaDocs.

**Review by venuslimm — APPROVED** [review]
LGTM!

**Pull request merged**